extension/src/test/suite/AutoIndent.test.ts

```diff
@@ -0,0 +1,101 @@
+import * as chai from "chai";
+import * as vscode from "vscode";
+import * as fs from "fs";
+import * as path from "path";
+import {findContainers,getIndentation,} from "../../format/autoIndent";
+import { ReadonlyDocument } from "../../project/readOnlyDocument";
+let assert = chai.assert;
+const testDir = path.join(__dirname + "/../../../extension/src/test");
+const outputDir = path.join(testDir + "/OutputFile");
+let indentTestFile = "";
+let indentTestDoc: vscode.TextDocument;
+
+fs.mkdir(outputDir, { recursive: true }, (err) => {
+  if (err) {
+    return console.error(err);
+  }
+});
+
+function createFakeTextDcoument() {
+  indentTestFile = path.join(testDir + "/SourceFile/autoIndent.lsp");
+  try {
+    if (fs.existsSync(indentTestFile)) {
+      indentTestDoc = ReadonlyDocument.open(indentTestFile);
+    }
+  } catch (error) {
+    console.log(error);
+  }
+}
+function getIndent(cursorPos2d: vscode.Position) {
+  try {
+    let containerElement = findContainers(indentTestDoc,cursorPos2d);
+    let lineIndentSpace = getIndentation(indentTestDoc,containerElement,cursorPos2d);
+    return lineIndentSpace.length;
+  } catch (error) {
+    console.log(error);
+  }
+}
+
+suite("Autoindent Tests", function () {
+  // Autoindent is used for input in the vscode editor
+  suiteSetup(() => {
+    createFakeTextDcoument();
+  });
+
+  test("Autoindent for defun enter should have 2 indentation", async function () {
```

Comment:
I don't know. 
From the lisp file above, defun has 0 indent and setq has 2 indent, but the test says both has 2. Is this expected?

Reply:
Yes, for both defun and setq under this context content it should have 2 indentation when press enter:
(defun c:indentTest (/ a b) 
 |(setq a 3)
 |